### ISSUE TITLE
Processing ERB in CSS files

### ISSUE THREAD

**Issue by charliesneath** [jekyll/jekyll-assets#205]
I'm trying to get background image assets working with Jekyll Assets and following the discussion about using the .erb extension to convert .erb tags, instead of relying on Liquid ([here](https://github.com/jekyll/jekyll-assets/issues/6)).

If `background.scss` contains `background: url('<%= asset_path image.png %>')`, should its filepath be `background.css.scss.erb`?

I'm getting hung up because when I `@import background` into `main.css` I get an error that `background.css` isn't found:

`Conversion error: File to import not found or unreadable: background`

Does anything addiitonal need to be added to `_config.yml` to make sure `background.css.scss.erb` is processed by ERB?

Thanks for the help! Hoping this gives me a better understanding for how Liquid and ERB fit into the Jekyll's asset pipeline.

(cc @ixti)

**Comment by envygeeks** [comment]
What version of Jekyll Assets are you using? If it's 1.x we no longer provide bug fixes for that.  If it's for 2.x we do not support erb at all as that goes against our goal of being included in Github pages.

**Comment by charliesneath** [comment]
@ixti Thanks for quick response.

I'm using latest version (which I think would be 2.x). 

That makes sense re GitHub pages. Does that mean that (for now) there isn't a way to process tags like `{% asset_path background.png %}` when they're included in CSS?

**Comment by envygeeks** [comment]
For your case you don't need that, we provide helpers: `asset_path(image.png)` will produce either a full URL (http) if you have a CDN or a baseurl+asset as an absolute path if you do not, with the digested path if you are in production or have enforced digesting.

**Comment by charliesneath** [comment]
Excellent, thanks. 

If I use a helper like `asset_path(image.png)`, where do I tell Jekyll that it needs to process the file?

Do I need to list the filename in `_config.yml` under `assets` or `source`?

Thank you!

**Comment by envygeeks** [comment]
You can't, we don't process any of your files with Liquid or ERB and probably won't in the foreseeable future, if you want a user configurable style my recommendation for now is to create a variables.scss file and reference that, at that point you've done the same thing natively to the pipeline.

To note my plans not to implement Liquid in the foreseeable future isn't a plan not to implement it, it's just something that's not necessarily important because you can create easy to edit variable files in SCSS/Less and most things in the pipeline minus images.

**Comment by charliesneath** [comment]
Got it, thanks.

The only thing I'm hung up on is getting `asset_path(background-test.jpg)` to produce the URL as you mentioned above. I don't have it working yet — how does the pipeline know to process the file with this code?

**Comment by envygeeks** [comment]
It does not produce a url, it only produces a path, you still need to wrap it in `url()`

**Comment by charliesneath** [comment]
I have the following line in `background.scss`:

`background: url('asset_path(background-test.jpg)');`.

Looking in the compiled `_site` directory, I still see the following: 

`background: url('asset_path(background-test.jpg)');`

From what you've described above, I would expect it to output the following:

`background: url('/assets/background-test.jpg');`

Any ideas why? Thanks for helping sort this out.

**Comment by envygeeks** [comment]
Because you put asset_path into quotes and made it a string literal.  Unquote it.

**Comment by charliesneath** [comment]
Removing the quotes did not fix the problem. The line is still outputting as 

`background: url(asset_path(background-test.jpg));`

These are my current settings in `_config.yml`:

```
assets:
  compress:
  - css: true
  digest: true
  assets:
    - "*.css"
    - "*.jpg"
  sources:
    - "_assets/"
    - "_sass/"
```

**Comment by envygeeks** [comment]
That doesn't help, provide more information or a repository I can clone and look at.  If the asset stays the same then the asset doesn't exist.  If the asset does exist and it doesn't display then I'll need to see the repo if available otherwise I can't explain why that would happen unless you can provide reliable reproduction tests.  As that is a surface we cover in the tests.

**Comment by charliesneath** [comment]
Moved image to `_assets/images` but seeing no change in functionality.

https://github.com/charliesneath/asset-test/commit/1bbede28f19f25ea0dbbbba242db97f9203bbf03

Do you have working example?

**Comment by envygeeks** [comment]
Yes, they are called specs, that and the hundreds of people who use the software.  The burden on me isn't to prove it works, it's the opposite, I'm to assume it works if 1: my tests are passing and 2: nobody else is reporting a problem.

I will try your source later tonight and see if I can replicate it on my system.

**Comment by envygeeks** [comment]
I knew that case was gonna come up when I was working on the code today and I hoped it wouldn't.  That's a bug, I need to append your sources not replace *our* sources.  I'm gonna leave this open and issue a bug fix today for 2.0.1.

**Comment by charliesneath** [comment]
That makes sense. What became unclear is whether the sources in the configuration for `_config.yml` listed in the [README.doc](https://github.com/jekyll/jekyll-assets#configuration) ( `_assets/css`, `_assets/images`, etc.) are set by default, or just an example of how you could manually define that your project's `_config.yml` file.
